### PR TITLE
3 create theme

### PR DESCRIPTION
**Pull request by Lovemore-Gakava**
feat: Added theme for charts and also figure 2 - pyramid_chart.

### CODE REVIEW

File: R/demography.R
Comment (chen-chen-stat) [inline R/demography.R:1]: @Lovemore-Gakava TBD: Do we want to document all example datasets in one file or document each example dataset in a separate file?

File: R/utils.R [hunk 1]
```diff
@@ -28,6 +70,276 @@ colfun <- function() {
...
+#' * **rel** - for legend elements.
+#'
+#' @export
+#' @examples
+#' control_fonts(base_font_size = 10)
+control_fonts <- function(base_font_size = 9,
+                          h1 = 12,
+                          h2 = 10,
+                          label = base_font_size + 1) {
+  rel <- 7.253 / 9
```
Comment (chen-chen-stat) [inline R/utils.R:114]: Should it be `rel <- 7.253 / base_font_size` here, or it does not change with base_font_size?

File: R/utils.R [hunk 2]
```diff
@@ -28,6 +70,276 @@ colfun <- function() {
...
+#' @param axis_title_y labels of axes
+#' @param axis_text_y_left tick labels along axes
+#' @param legend_position position
+#' @param panel_grid_minor grid lines
+#' @param panel_grid_major grid lines
+#' @param ... Additional arguments passed to other methods.
+#'
+#' @return theme for chart
+#' @export
+charts_style_theme <- function(base_family = "",
```
Comment (chen-chen-stat) [inline R/utils.R:144]: BR_charts_theme

File: R/pyramid_chart.R [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#' Demography Figure
```
Comment (chen-chen-stat) [inline R/pyramid_chart.R:1]:
```suggestion
#' Pyramid Chart
```

File: R/pyramid_chart.R [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+#'
+#' @param fig2_data `dataframe` demography data
```
Comment (chen-chen-stat) [inline R/pyramid_chart.R:3]: We may consider avoiding the use of figX in the codes.

File: R/pyramid_chart.R [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+#' @param Typev `Factor` Type of Diabetis A or B
+#' @param gendervar `Factor` Gender variable
```
Comment (chen-chen-stat) [inline R/pyramid_chart.R:5]: Do we want to make this more general? e.g., Factor 1, Factor 2.

File: R/utils.R [hunk 6]
```diff
@@ -5,6 +5,47 @@
...
+      5,
+      6,
+      7,
+      8,
+      9,
+      10,
+      11
+    )
+  )
   fig2_colors <- c("#00AFBB", "#FFDB6D")
```
Comment (chen-chen-stat) [inline R/utils.R:49]: We may consider renaming these to avoid the use of figX in the codes.

**Comment by chen-chen-stat** [comment]
@Lovemore-Gakava Hi LG, thank you for creating the PR. I have completed my review and added my comments for consideration. I am happy to discuss further if needed.

File: R/demography.R
Comment (Lovemore-Gakava) [reply R/demography.R:1]: @chen-chen-stat - agreed will create script data.R as suggested here: https://bookdown.org/rdpeng/RProgDA/documentation.html

File: R/utils.R (hunk 1 above)
Comment (Lovemore-Gakava) [reply R/utils.R:114]: Thanks Chen - line updated.

File: R/pyramid_chart.R (hunk 4 above)
Comment (Lovemore-Gakava) [reply R/pyramid_chart.R:3]: Agreed - we can discuss further.

File: R/pyramid_chart.R (hunk 5 above)
Comment (Lovemore-Gakava) [reply R/pyramid_chart.R:5]: Updated to add in documentation that the param shows levels of factor.

File: R/pyramid_chart.R [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      limits = ~figlimits,
+      breaks = seq(figlimits[1], figlimits[2], 10),
+      labels = abs(seq(figlimits[1], figlimits[2], 10))
+    )
+
+    fig2_1 <- fig2_data |>
+      filter(.data[["Type"]] == Typev[[1]]) |>
+      ggplot(aes(
+        x = .data[["figprev"]],
+        y = .data[["Age"]],
```
[the remark this answers is not in the document]
Comment (Lovemore-Gakava) [reply R/pyramid_chart.R:42]: added parameters for x-axis and yaxis inputs.

File: R/utils.R (hunk 6 above)
Comment (Lovemore-Gakava) [reply R/utils.R:49]: for the color function we can leave as is for now as we are generating the figures for the manuscript. LWe can consider the colors to be controlled via an argument in the function?

File: R/data.R [hunk 8]
```diff
@@ -18,3 +18,21 @@
 #' @docType data
 #' @format A data frame with 105 rows and 51 variables
 "brdata"
+
+
+#' Example demography data
+#'
+#' Simulated demography data
```
Comment (chen-chen-stat) [inline R/data.R:25]:
```suggestion
```

File: R/data.R [hunk 9]
```diff
@@ -18,3 +18,21 @@
...
+#'
```
Comment (chen-chen-stat) [inline R/data.R:26]:
```suggestion
```

File: R/data.R [hunk 10]
```diff
@@ -18,3 +18,21 @@
...
+#'
+#' @format ## `who`
```
Comment (chen-chen-stat) [inline R/data.R:28]:
```suggestion
#' @format ## `demography`
```

**Review by chen-chen-stat — APPROVED** [review]

**Pull request merged**